core/lib/libtomcrypt/cipher.c

```diff
@@ -0,0 +1,623 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2014-2019, Linaro Limited
+ */
+
+#include <assert.h>
+#include <crypto/crypto.h>
+#include <crypto/crypto_impl.h>
+#include <stdlib.h>
+#include <string.h>
+#include <tee_api_types.h>
+#include <tee/tee_cryp_utl.h>
+#include <tomcrypt.h>
+#include <utee_defines.h>
+#include <util.h>
+
+/******************************************************************************
+ * Symmetric ciphers
+ ******************************************************************************/
+
+static void __maybe_unused get_des2_key(const uint8_t **key, size_t *key_len,
+					uint8_t *tmp)
+{
+	if (*key_len == 16) {
+		/*
+		 * This corresponds to a 2DES key. The 2DES encryption
+		 * algorithm is similar to 3DES. Both perform and
+		 * encryption step, then a decryption step, followed
+		 * by another encryption step (EDE). However 2DES uses
+		 * the same key for both of the encryption (E) steps.
+		 */
+		memcpy(tmp, *key, 16);
+		memcpy(tmp + 16, *key, 8);
+		*key = tmp;
+		*key_len = 24;
+	}
+}
+
+#if defined(CFG_CRYPTO_ECB)
```

Comment:
...and split files? (that get_des2_key() is annoying, though, could it be inlined in a `.h`?)